guides/Advanced SettingGateway/UnderstandingSchemaFolders.md

```diff
@@ -0,0 +1,164 @@
+
+You would normally use these two methods using the following snippet:
+
+```javascript
+// Add a new key
+this.client.gateways.gatewayName.schema.addKey(name, options, force);
+
+// Add a new folder
+this.client.gateways.gatewayName.schema.addFolder(name, options, force);
+```
+
+The parameters are:
+
+- **name**: The name of the new key. If it conflicts with a pre-existent key, this will error.
+- **options**: The options for the new key or folder.
+- **force**: Whether this change should affect all entries. It requires a lot of processing but ensures the changes are correctly applied in both cache and database.
+
+You can also extend any of the three built-in {@link Gateway}s from Klasa, for example, if you want to add a new key called **modlogs** that accepts only text channels, for your guild configs, you would use the following code:
+
+```javascript
+this.client.gateways.guilds.schema.addKey('modlogs', { type: 'TextChannel' });
+```
+
+Where you're doing the following steps:
+
+1. Access to {@link KlasaClient#gateways}, type of {@link GatewayDriver}, which holds all gateways.
+1. Access to the guilds' {@link Gateway}, which manages the per-guild configuration.
+1. Access to the guilds' schema via {@link Gateway#schema}, which manages the gateway's schema.
+1. Add a new key called **modlogs** in the root of the schema, with type of **TextChannel**.
+
+And you would have a perfectly configured modlogs key in your configs. However, you can also have an array of the same type. For example, you want to have a configurable array of users blacklisted in a guild, in a key named **userBlacklist**:
+
+```javascript
+this.client.gateways.guilds.schema.addKey('userBlacklist', { type: 'User', array: true });
+```
+
+And now you can access to any of them in your guild configs like in the following snippet!
+
+```javascript
+msg.guild.configs.modlogs;
+// null
+msg.guild.configs.userBlacklist;
+// []
+```
+
+## Removing keys
+
+Removing keys with the schema is quite easy, as you would access to the {@link SchemaFolder} that holds it and remove it by its name (remember that `force` is optional and defaults to `true`) using {@link SchemaFolder#removeKey} as the following example:
+
+```javascript
+this.client.gateways.gatewayName.schema.removeKey(name, force);
+```
+
+In case you have a key you do not longer use and you want to get rid of it, for example, the recently created **userBlacklist** key for guild configs, you would run the following code:
+
+```javascript
+this.client.gateways.guilds.schema.removeKey('userBlacklist');
+```
+
+And the property `userBlacklist` for all guild configs will be deleted, that being said:
+
+```javascript
```

Comment:
I don't think it is necessary to have prove that the key no longer exists in that code snippet, I feel it is good enough saying that the key will no longer exist